target/linux/octeon/base-files/lib/preinit/03_rename_interfaces

```diff
@@ -0,0 +1,15 @@
+rename_interfaces() {
+	. /lib/functions/system.sh
+	. /lib/functions.sh
+
+	case "$(board_name)" in
+		ubnt,edgerouter-4)
+			rename_interface_by_mac lan0 $(macaddr_add $(mtd_get_mac_binary eeprom 0) +1)
+			rename_interface_by_mac lan1 $(macaddr_add $(mtd_get_mac_binary eeprom 0) +2)
+			rename_interface_by_mac lan2 $(macaddr_add $(mtd_get_mac_binary eeprom 0) +3)
+			rename_interface_by_mac lan3 $(mtd_get_mac_binary eeprom 0)
```

Comment:
What's the actual reason for doing a "rename" of interfaces here? What's wrong with the default?

Reply:
Without this, the order does does not correspond to the labels on the device, which is very confusing. Any other attempts to fix this in the DTS failed so far.

Reply:
But why that complicated detour with the MAC addresses? Can't we just rename eth3 to lan2, eth9 to lan1 etc.?
So, instead of determining the names by the MAC addresses _every_ time, just determine it once and put the result here?

Reply:
because we need a resilient solution and currently kernel does not guarantee interface naming order.

maybe we could do that if we opt for https://en.wikipedia.org/wiki/Consistent_Network_Device_Naming but openwrt does not use that.

renaming by mac address is much more resilient solution.
from my experience it is even more resilient then that 'consistent network device naming' from kernel. i have had devices change names between kernel releases or reboots (because of device initialization order) but mac addresses obviously still stay the same.

Reply:
Have you tried already to just patch in a label like done recently for mt7621/DSA:
https://github.com/openwrt/openwrt/commit/3211c983fd7a51208f22866193df64a009b64fdb

Reply:
thanks, just added a patch to do it via device tree labels